WebApplication/Controllers/DownloadController.cs

```diff
@@ -39,23 +42,35 @@ public DownloadController(ILogger<DownloadController> logger, UserResolver userR
         }
 
         [HttpGet("{projectName}/{hash}/model/{token?}")]
-        public Task<RedirectResult> Model(string projectName, string hash, string token=null)
+        public Task<RedirectResult> Model(string projectName, string hash, string token = null)
         {
-            if (token != null)
-                _userResolver.Token = token;
+            EnsureToken(token);
 
-            return RedirectToOssObject(projectName, hash, (ossNames, isAssembly ) => ossNames.GetCurrentModel(isAssembly));
+            return RedirectToOssObject(projectName, hash, (ossNames, isAssembly) => ossNames.GetCurrentModel(isAssembly));
         }
 
         [HttpGet("{projectName}/{hash}/rfa/{token?}")]
         public Task<RedirectResult> RFA(string projectName, string hash, string token = null)
         {
-            if (token != null)
-                _userResolver.Token = token;
+            EnsureToken(token);
 
             return RedirectToOssObject(projectName, hash, (ossNames, _)=> ossNames.Rfa);
         }
 
+        [HttpGet("{projectName}/{hash}/bom/{token?}")]
```

Comment:
Can be the reading done by our TokenHandler middleware? I think it can take a look at the specific query parameter. It would make code much more concise.

Reply:
Good idea!  Let's move it to separate PR?